src/scripts/main.js

```diff
@@ -1,3 +1,330 @@
+  },
+
+  {
+    label: 'Age:',
+    name: 'age',
+    type: 'number',
+    qa: 'age',
+  },
+
+  {
+    label: 'Salary:',
+    name: 'salary',
+    type: 'number',
+    qa: 'salary',
+  },
+];
+
+inputs.forEach((input) => {
+  const label = document.createElement('label');
+
+  label.textContent = input.label;
+
+  const inputElement = document.createElement('input');
+
+  inputElement.name = input.name;
+  inputElement.type = input.type;
+  inputElement.setAttribute('data-qa', input.qa);
+
+  label.appendChild(inputElement);
+  form.appendChild(label);
+  form.appendChild(document.createElement('br'));
+});
+
+// creating select for office
+
+const officeLabel = document.createElement('label');
+
+officeLabel.textContent = 'Office:';
+
+const select = document.createElement('select');
+
+select.name = 'office';
+select.setAttribute('data-qa', 'office');
+
+[
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+].forEach((city) => {
+  const option = document.createElement('option');
+
+  option.value = city;
+  option.textContent = city;
+  select.appendChild(option);
+});
+
+officeLabel.appendChild(select);
+form.appendChild(officeLabel);
+form.appendChild(document.createElement('br'));
+
+document.body.appendChild(form);
+
+// creating button
+
+const submitButton = document.createElement('button');
+
+submitButton.type = 'submit';
+submitButton.textContent = 'Save to table';
+
+form.appendChild(submitButton);
+
+form.addEventListener('submit', function (e) {
+  e.preventDefault();
+
+  // Get form values
+  const formName = form.querySelector('input[name="name"]').value.trim();
+  const formPosition = form
+    .querySelector('input[name="position"]')
+    .value.trim();
+  const formAge = parseInt(
+    form.querySelector('input[name="age"]').value.trim(),
+    10,
+  );
+  const formSalary = parseInt(
+    form
+      .querySelector('input[name="salary"]')
+      .value.trim()
+      .replace(/[$,]/g, ''),
+    10,
+  );
+  const formOffice = form.querySelector('select[name="office"]').value;
+
+  // Validate all fields are filled
+  if (
+    !formName ||
+    !formPosition ||
+    !formOffice ||
+    isNaN(formAge) ||
+    isNaN(formSalary)
+  ) {
+    createNotification('All fields are required.', 'error');
+
+    return;
+  }
+
+  // Validate Name: must be at least 4 characters long
+  if (formName.length < 4) {
+    createNotification('Name must be at least 4 characters long.', 'error');
+
+    return;
+  }
+
+  // Validate Age: must be between 18 and 90
+  if (formAge < 18 || formAge > 90) {
+    createNotification('Age must be between 18 and 90.', 'error');
+
+    return;
+  }
+
+  // Create new row in the table
+  const newRow = document.createElement('tr');
+
+  newRow.innerHTML = `
+  <td>${formName}</td>
+  <td>${formPosition}</td>
+  <td>${formOffice}</td>
+  <td>${formAge}</td>
+  <td>$${formSalary.toLocaleString()}</td>
+  `;
+
+  tbody.appendChild(newRow);
+  rows.push(newRow);
```

Comment:
Pushing the new row to the `rows` array here will not update the array used in the sorting functions. Consider re-initializing the `rows` array after adding a new row to ensure it includes all table rows.